### DISCUSSION TITLE
Issue when changing prisma client location

### DISCUSSION THREAD

**Discussion by DrewVartanian** [prisma/prisma#20630] [Q&A]
### Question

I am trying to change the location of my generated Prisma client, so that I can check it into the codebase and use the generated client/types when building and deploying my project, but am running into some issues.  In my schema.prisma file, I changed:
```prisma
 generator client {
   provider        = "prisma-client-js"
   previewFeatures = ["postgresqlExtensions", "filteredRelationCount"]
 }

 generator pothos {
   provider     = "prisma-pothos-types"
   clientOutput = "@prisma/client"
   output       = "./generated.ts"
 }
```
to 
```prisma
 generator client {
   provider        = "prisma-client-js"
   output          = "./client"
   previewFeatures = ["postgresqlExtensions", "filteredRelationCount"]
 }

 generator pothos {
   provider     = "prisma-pothos-types"
   clientOutput = "./client"
   output       = "./generated.ts"
 }
```
I'm using next.js, and have @next-auth/prisma-adapter in my [...nextauth].tsx connecting the prisma client to NextAuth:
```ts
import NextAuth from 'next-auth';
import { PrismaAdapter } from '@next-auth/prisma-adapter';
import { PrismaClient } from 'src/prisma/client';
...
const prisma = new PrismaClient();
...
export default NextAuth({
  adapter: PrismaAdapter(prisma),
...
```
but am getting a typescript error at that last line saying:
```
Type error: Argument of type 'PrismaClient<PrismaClientOptions, never, DefaultArgs>' is not assignable to parameter of type 'PrismaClient<PrismaClientOptions, never, RejectOnNotFound | RejectPerOperation | undefined>'.
  Types of property '$use' are incompatible.
    Type '(cb: import("/Users/drew/paloma/src/prisma/client/index").Prisma.Middleware<any>) => void' is not assignable to type '(cb: import("/Users/drew/paloma/node_modules/.prisma/client/index").Prisma.Middleware<any>) => void'.
      Types of parameters 'cb' and 'cb' are incompatible.
        Types of parameters 'params' and 'params' are incompatible.
          Type 'import("/Users/drew/paloma/src/prisma/client/index").Prisma.MiddlewareParams' is not assignable to type 'import("/Users/drew/paloma/node_modules/.prisma/client/index").Prisma.MiddlewareParams'.
            Types of property 'action' are incompatible.
              Type 'import("/Users/drew/paloma/src/prisma/client/index").Prisma.PrismaAction' is not assignable to type 'import("/Users/drew/paloma/node_modules/.prisma/client/index").Prisma.PrismaAction'.
                Type '"findFirstOrThrow"' is not assignable to type 'PrismaAction'.
```
When I trying running on my dev anyways, I get the following error when I load a page that hits the database through prisma:
```
ApolloError: 
Invalid `prisma.$queryRaw()` invocation:


Raw query failed. Code: `42883`. Message: `db error: ERROR: operator does not exist: text = jsonb
HINT: No operator matches the given name and argument types. You might need to add explicit type casts.`
```
I wasn't getting any of these errors before I changed my schema.prisma.  Does anyone have any clues on what might be going wrong here?

### Information about Prisma Schema, Client Queries and Environment (optional)

- OS: macOS 13.5
- Database: PostgreSQL
- Node.js version: v18.17.0
- prisma version: 5.1.1
- @prisma/client version: 5.1.1
- typescript version: 5.1.6
- next version: 13.1.2
- next-auth version: 4.22.1
- @next-auth/prisma-adapter version: 1.0.7
- @pothos/core: 3.30.0
- @pothos/plugin-prisma: 3.51.1

**Comment by ludralph** [comment]
Hi @DrewVartanian 👋 

Can you please verify that you're correctly importing `PrismaClient` from the correct `clientOutput` path? 

According to the `Pothos-GraphQL` documentation, the 
> `clientOutput`: Where the generated code will import the PrismaClient from. The default is the full path of wherever the client is generated. If you are checking in the generated file, using @prisma/client is a good option.

You might also want to share a minimal repository so I can take a look.

**Reply by krzysztof-fijolek** [reply]
Hi @DrewVartanian 

Did you manage to resolve this issue? I have encountered this as well. The project runs and build just fine on local. However some - not all - of Vercel deploys fail. Re-deploys from the same commit usually fixes the problem.

**Reply by faithandjs** [reply]
Just redeployed after seeing your comment and it deployed after failing several times, did you get a fix that lets it auto deploy?

**Reply by poweriton** [reply]
You just saved me there. I would assume rebuild started a fresh build instance, or does it resume an existing one? This feels related to a Prisma client generation issue

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
